data-upgrade-processes-documents/src/main/java/org/exoplatform/migration/ProcessesDocumentsUpgradePlugin.java

```diff
@@ -0,0 +1,259 @@
+import org.exoplatform.services.jcr.RepositoryService;
+import org.exoplatform.services.jcr.ext.common.SessionProvider;
+import org.exoplatform.services.jcr.impl.core.NodeImpl;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.container.xml.InitParams;
+
+import javax.jcr.ItemExistsException;
+import javax.jcr.ItemNotFoundException;
+import javax.jcr.Node;
+import javax.jcr.NodeIterator;
+import javax.jcr.RepositoryException;
+import javax.jcr.Session;
+import javax.jcr.UnsupportedRepositoryOperationException;
+import javax.jcr.query.Query;
+import javax.jcr.query.QueryManager;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.net.URLDecoder;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ProcessesDocumentsUpgradePlugin  extends UpgradeProductPlugin {
+  private static final Log      log = ExoLogger.getExoLogger(ProcessesDocumentsUpgradePlugin.class);
+
+  private RepositoryService       repositoryService;
+  private OnlyofficeEditorService onlyofficeEditorService;
+  private EntityManagerService entityManagerService;
+  private UserACL              userACL;
+
+  public ProcessesDocumentsUpgradePlugin(EntityManagerService entityManagerService, RepositoryService repositoryService, OnlyofficeEditorService onlyofficeEditorService, UserACL userACL, InitParams initParams) {
+    super(initParams);
+    this.repositoryService = repositoryService;
+    this.onlyofficeEditorService = onlyofficeEditorService;
+    this.userACL = userACL;
+    this.entityManagerService = entityManagerService;
+  }
+
+  @Override
+  public void processUpgrade(String oldVersion, String newVersion) {
+    long startupTime = System.currentTimeMillis();
+    log.info("Start Processes documents upgrade");
+
+    List<String> errorPath = new ArrayList<>();
+
+    try {
+
+
+      List<AttachmentContextEntity> attachments = getAllAttachments();
+
+      SessionProvider sessionProvider = SessionProvider.createSystemProvider();
+      Session session = sessionProvider.getSession("collaboration", repositoryService.getDefaultRepository());
+
+      Map<String,String> attachmentIdReplacement = new HashMap<>();
+
+      long total = attachments.size();
+      int currentIndex=0;
+      log.info("Processes documents upgrade Select query executed in {} ms, {} elements to check", System.currentTimeMillis()-startupTime, total);
+
+      for (AttachmentContextEntity attachment : attachments) {
+        if (!attachmentIdReplacement.containsKey(attachment.getAttachmentId())) {
+
+          try {
+            Node file = session.getNodeByUUID(attachment.getAttachmentId());
+
+
+            log.debug("Proceed node {}", file.getPath());
+
+            if (!file.getPath().startsWith("/Trash") && file.isNodeType("nt:file") && (file.getName().endsWith(".docxf")
```

Comment:
IMO it will be better to check if the file is not null

Reply:
file == null means that session.getNodeByUUID raise a ItemNotFoundException which is already catched. So for me no need to check nullity